tools/backup_analysis.py

```diff
@@ -0,0 +1,174 @@
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# Compare WT data files in two different home directories.
+
+import fnmatch, glob, os, sys, time
+
+def usage_exit():
+    print('Usage: backup_analysis.py dir1 dir2 [granularity]')
+    print('  dir1 and dir2 are POSIX pathnames to WiredTiger backup directories,')
+    print('Options:')
+    print('  granularity - an (optional) positive integer indicating the granularity')
+    print('  of the incremental backup blocks. The default is 16MB')
+    sys.exit(1)
+
+def die(reason):
+    print('backup_analysis.py: error: ' + reason, file=sys.stderr)
+    sys.exit(1)
+
+# Check that the directory given is a backup. Look for the WiredTiger.backup file.
+def check_backup(mydir):
+    if not os.path.isdir(mydir):
+        return False
+    backup_file=mydir + "/WiredTiger.backup"
+    if not os.path.exists(backup_file):
+        return False
+    return True
+
+def compare_file(dir1, dir2, filename, file_size, cmp_size, granularity):
+    # Initialize all of our counters per file.
+    bytes_gran = 0
+    gran_blocks = 0
+    num_blocks = file_size // cmp_size
+    partial = file_size % cmp_size
+    pct20 = granularity // 5
+    pct80 = pct20 * 4
+    pct20_count = 0
+    pct80_count = 0
+    start_off = 0
+    total_bytes_diff = 0
+    fp1 = open(os.path.join(dir1, filename), "rb")
+    fp2 = open(os.path.join(dir2, filename), "rb")
+    # Time how long it takes to compare each file.
+    start = time.asctime()
+    # Compare the bytes in cmp_size blocks between both files.
+    for b in range(0, num_blocks + 1):
+        offset = cmp_size * b
+        # Gather and report information when we cross a granularity boundary.
+        if offset % granularity == 0:
+            if offset != 0 and bytes_gran != 0:
+                print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
```

Comment:
Instead of `format` function, the cool kids these days are using f-strings: https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings
Like:
```
    print(f'{filename}: Offset {start_off}: {bytes_gran} bytes differ in {granularity} bytes')
```
You can use f'...' or f"...", same thing.